Add section 3.5 to Docusaurus (#91)
iotaledger/IOTA-2.0-Research-Specifications: docs/3.5 Manual Peering.md

# 3.5 Manual Peering

> Manual peering enables node operators to manually manage nodes' neighbors, orthogonal to the peering protocols described in [Section 3.3 - Peer Discovery](./3.3%20Peer%20Discovery.md) and [Section 3.4 - Peer Selection](./3.4%20Peer%20Selection.md). This section defines manual peering protocol.

Reviewer: The links don't work in docusarus
```suggestion
Manual peering enables node operators to manually manage nodes' neighbors, orthogonal to the peering protocols described in [Section 3.3 - Peer Discovery](./3.3%20Peer%20Discovery) and [Section 3.4 - Peer Selection](./3.4%20Neighbor%20Selection). This section defines manual peering protocol.
```

## 3.5.1 Motivation

> Manual peering can be additional protection against eclipse attacks as the manual peering is completely in the hands of the node operator based on real-world trust. Furthermore, it allows operating nodes without exposing their IP addresses to the network.

Reviewer 2: ```suggestion
Manual peering can be additional protection against eclipse attacks as the manual peering is completely in the hands of the node operator based on real-world trust. Furthermore, it allows operating nodes to function without exposing their IP addresses to the network.
```

> ## 3.5.2 Detailed design

Reviewer 2: ```suggestion
## 3.5.2 Detailed Design
```

> The known peers are given by the node operator, containing peers' **public key** and **IP address**. They are divided into two groups by comparing their public keys to the node's public key lexicographically:

Reviewer 2: ```suggestion
The known peers are given by the node operator, containing the peers' **public key** and **IP address**. They are divided into two groups by comparing their public keys to the node's public key lexicographically:
```

* Chosen neighbors (outbound): If `node < peer`, the node proactively sends a peering request to the peer.
* Accepted neighbors (inbound): If `node > peer`, the node awaits the peer to send a peering request and accept it.

> After the group of peers is decided, the node shall either send peering requests to the chosen neighbors or wait for peering requests from accepted neighbors. The connections are checked every `reconnectInterval`, if a connection is dropped, the node shall try to re-establish it until the peer is removed or the node shutdowns.

Reviewer 2: ```suggestion
After the group of peers is decided, the node shall either send peering requests to the chosen neighbors or wait for peering requests from accepted neighbors. The connections are checked at every `reconnectInterval`, if a connection is dropped, the node shall try to re-establish it until the peer is removed or the node shutdowns.
```

> A node removes a peer by closing the connection directly without sending a drop message described in [Section 3.4 - Peer Selection](./3.4%20Neighbor%20Selection) to the peer. Thus, the removed peer shall still try to reconnect to the node every `reconnectInterval`, until its operator  removes the node from the manual peering list. That is, it is the node operators' responsibility to monitor and manage the known peer list in manual peering.

Reviewer 2: ```suggestion
A node removes a peer by closing the connection directly without sending a drop message, described in [Section 3.4 - Peer Selection](./3.4%20Neighbor%20Selection), to the peer. Thus, the removed peer shall still try to reconnect to the node at every `reconnectInterval`, until its operator removes the node from the manual peering list. That is, it is the node operators' responsibility to monitor and manage the known peer list in manual peering.
```

> A node removes a peer by closing the connection directly without sending a drop message described in [Section 3.4 - Peer Selection](./3.4%20Peer%20Selection.md) to the peer. Thus, the removed peer shall still try to reconnect to the node every `reconnectInterval`, until its operator  removes the node from the manual peering list. That is, it is the node operators' responsibility to monitor and manage the known peer list in manual peering.

Reviewer: ```suggestion
A node removes a peer by closing the connection directly without sending a drop message described in [Section 3.4 - Peer Selection](./3.4%20Neighbor%20Selection) to the peer. Thus, the removed peer shall still try to reconnect to the node every `reconnectInterval`, until its operator  removes the node from the manual peering list. That is, it is the node operators' responsibility to monitor and manage the known peer list in manual peering.
```

> ## 3.5.3 Request and response

Reviewer 2: ```suggestion
## 3.5.3 Request and Response
```

Connections among manual peers are TCP connections. After the connection is established, nodes shall perform handshakes to validate the connected peer. Each handshake request and response shall be encapsulated into a `data` and signed with the ed25519 private key of the sender's [identity](#Node_identities) that contains the related public key to allow the receiver to verify the signature. All the received requests and responses shall be verified and those with invalid version numbers and signatures be discarded.

### Request and Response Layout
<table>
     <tr>
         <th>Name</th>
         <th>Type</th>
         <th>Description</th>
     </tr>
     <tr>
         <td>data</td>
         <td>ByteArray</td>
         <td>Contains the handshake request/response.</td>
     </tr>
     <tr>
         <td>public_key</td>
         <td>ByteArray[32]</td>
         <td>The ed25519 public key of the peer's identity used to verify its signature.</td>
     </tr>
     <tr>
         <td>signature</td>
         <td>ByteArray[32]</td>
         <td>The ed25519 signature of the `data` field, signed by using the private key of the peer's identity.</td>
     </tr>
 </table>

> ### Handshake request

Reviewer 2: ```suggestion
### Handshake Request
```
